migrations/0002.sqlcode.sql

```diff
@@ -0,0 +1,90 @@
+-- We want to re-create the procedure DropCodeSchema.
+-- Frist, everything related to this procedure must be dropped
+-- before it is re-created in the end.
+
+drop procedure sqlcode.DropCodeSchema;
+
+create procedure sqlcode.DropCodeSchema(@schemasuffix varchar(50))
+as begin
+    set xact_abort, nocount on
+    begin try
+        declare @msg varchar(max)
+        declare @sql nvarchar(max)
+
+        if @@trancount = 0 throw 55001, 'You should run sqlcode.CreateCodeSchema within a transaction', 1;
+
+        declare @schemaname nvarchar(max) = concat('code@', @schemasuffix)
+        declare @schemaid int = (select schema_id from sys.schemas where name = @schemaname);
+        if @schemaid is null
+        begin
+            set @msg = concat('Schema [code@', @schemasuffix, '] not found');
+            throw 55002, @msg, 1;
+        end
+
+        -- Drop views, functions, procedures
+	declare @curVFP cursor; -- VFP: views, functions, procedures
+	set @curVFP = cursor local read_only forward_only for
+	    select
+	        concat('drop ', v.DropType, ' ', quotename(@schemaname), '.', quotename(o.name))
+	    from sys.objects as o
+	    cross apply ( values ( case
+	        when o.type = 'FN' then 'function'
+	        when o.type = 'IF' then 'function'
+	        when o.type = 'TF' then 'function'
+	        when o.type = 'P' then 'procedure'
+	        when o.type = 'PC' then 'procedure'
+	        when o.type = 'V' then 'view'
+	    end )) v(DropType)
+	    where o.schema_id = @schemaid and v.DropType is not null;
+
+	open @curVFP
+
+	while 1 = 1
```

Comment:
This will loop forever, then there's an error , and this probably crashes and the rest doesn't execute.

You should instead check `@@fetch_status`. This should do the trick:

```
fetch next from @curVFP into @sql;
while (@@fetch_status = 0)
begin
    exec sp_executesql @sql;
    fetch next from @curVFP into @sql;
end
```